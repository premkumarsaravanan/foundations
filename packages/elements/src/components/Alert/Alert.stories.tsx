import React from 'react'
import { storiesOf } from '@storybook/react'
import { Alert } from './index'

storiesOf('Alert', module)
  .add('Success', () => <Alert message="success" type="success" />)
  .add('Primary', () => <Alert message="primary" type="primary" />)
  .add('Info', () => <Alert message="info" type="info" />)
  .add('Danger', () => <Alert message="danger" type="danger" />)
