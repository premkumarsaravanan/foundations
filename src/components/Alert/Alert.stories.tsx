import React from 'react'
import { storiesOf } from '@storybook/react'
import { Alert } from './index'

storiesOf('Alert', module)
  .add('success', () => <Alert message="success" type="success" />)
  .add('primary', () => <Alert message="primary" type="primary" />)
  .add('info', () => <Alert message="info" type="info" />)
  .add('danger', () => <Alert message="danger" type="danger" />)
