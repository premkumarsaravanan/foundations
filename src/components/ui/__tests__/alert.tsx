import * as React from 'react'
import { shallow, mount } from 'enzyme'
import Alert from '../alert'
import toJson from 'enzyme-to-json'

describe('Alert', () => {
  it('should match a snapshot', () => {
    expect(toJson(shallow(<Alert message="Some random message" />))).toMatchSnapshot()
  })

  it('should us to change type props', () => {
    const wrapper = mount(<Alert message="Some random message" type="danger" />)
    expect(wrapper.props().type).toEqual('danger')
    wrapper.setProps({ type: 'warning' })
    expect(wrapper.props().type).toEqual('warning')
  })

  it('should allow custom className', () => {
    const wrapper = shallow(<Alert message="Some random message" className="addition" />)
    expect(wrapper.find('.addition')).toHaveLength(1)
  })

  it('should render close button if have closable prop', () => {
    const wrapper = shallow(<Alert message="Error message" closable />)
    expect(wrapper.find('.close')).toHaveLength(1)
  })

  it('simulates afterClose event', () => {
    const mockCallBack = jest.fn()
    const wrapper = shallow(<Alert message="Error message" afterClose={mockCallBack} closable />)
    wrapper.find('button').simulate('click')
    expect(mockCallBack).toBeCalledTimes(1)
  })

  afterEach(() => {
    jest.resetAllMocks()
  })
})
