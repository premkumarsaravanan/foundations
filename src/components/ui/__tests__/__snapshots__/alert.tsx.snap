// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should match a snapshot 1`] = `
<div
  className="alert alert-primary"
  role="alert"
>
  Some random message
</div>
`;
